### PR TITLE
chore: add pkg/api/node ut

### PR DESCRIPTION
**Pull request by yyzxw**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Improve package `pkg/api/node` unit test coverage.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by pacoxu** [comment]
The current code coverage for util.go is only 34.8%.

https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit&include-filter-by-regex=pkg/api/node

->

With this PR, the coverage is 100%. 👍

**Review by pacoxu — APPROVED** [review]
/lgtm
/assign @thockin

File: pkg/api/node/util.go [hunk 1]
```diff
@@ -62,6 +62,10 @@ func GetWarningsForNodeSelector(nodeSelector *metav1.LabelSelector, fieldPath *f
 		return nil
 	}
 
+	if fieldPath == nil {
```
Comment (thockin) [inline pkg/api/node/util.go:65]: Why would this ever legitmately be passed in as nil?

File: pkg/api/node/util.go [hunk 2]
```diff
@@ -89,6 +93,10 @@ func GetWarningsForNodeSelector(nodeSelector *metav1.LabelSelector, fieldPath *f
 
 // GetWarningsForNodeSelectorTerm checks match expressions of node selector term
 func GetWarningsForNodeSelectorTerm(nodeSelectorTerm api.NodeSelectorTerm, fieldPath *field.Path) []string {
+	if fieldPath == nil {
```
Comment (thockin) [inline pkg/api/node/util.go:96]: same

**Comment by thockin** [comment]
No updates on this - keep or close?

**Pull request closed without merging**